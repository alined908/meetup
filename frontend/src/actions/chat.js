import * as types from "../constants/action-types"
import {axiosClient} from '../accounts/axiosClient'

export const getRoom = (uri) => async dispatch => {
    dispatch({type: types.GET_ROOMS_REQUEST})
    try {
        const response = await axiosClient.get(
            `/api/chats/${uri}/`, {headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
            }}
        )
        dispatch({type: types.GET_ROOMS_SUCCESS, payload: response.data})
    } catch(e){
        dispatch({type: types.GET_ROOMS_ERROR, payload: "Unable to get chat room."})
    }
}   

export const getRooms = () => async dispatch => {
    dispatch({type: types.GET_ROOMS_REQUEST})
    try {
        const response = await axiosClient.get(
            "/api/chats/", {headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
            }}
        )
        dispatch({type: types.GET_ROOMS_SUCCESS, payload: response.data})
    } catch(e){
        dispatch({type: types.GET_ROOMS_ERROR, payload: "Unable to get chat rooms."})
    }
}

export const getMessages = (room) => async dispatch => {
    dispatch({type: types.GET_MESSAGES_REQUEST})
    try {
        const response = await axiosClient.get(
            `/api/chats/${room}/messages/`, {headers: {
            "Authorization": `Bearer ${localStorage.getItem('token')}`
        }})
        dispatch({type: types.GET_MESSAGES_SUCCESS, payload: response.data})
    } catch(e){
        dispatch({type: types.GET_MESSAGES_ERROR, payload: "Unable to get messages."})
    }
}

export const getMoreMessages = (room, last_message_id) => async dispatch => {
    dispatch({type: types.GET_MORE_MESSAGES_REQUEST})
    try {
        const response = await axiosClient.get(
            `/api/chats/${room}/messages/`, {params: {last: last_message_id}, headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
        }})
        dispatch({type: types.GET_MORE_MESSAGES_SUCCESS, payload: response.data})
    } catch(e){
        dispatch({type: types.GET_MORE_MESSAGES_ERROR, payload: "Unable to get more messages."})
    }

}

export const setActiveRoom = (room) => {
    return {
        type: types.SET_ACTIVE_ROOM,
        payload: {"uri": room}
    }
}

export const removeActiveRoom = () => {
    return {type: types.REMOVE_ACTIVE_ROOM}
}

export const updateRoom = (event) => async dispatch => {
    dispatch({type: types.UPDATE_ROOM, payload: event.message})
}

export const setTypingValue = (value) => {
    return {
        type: types.SET_TYPING_VALUE,
        payload: value
    }
}

export const addMessage = (message) => async dispatch => {
    dispatch({type: types.ADD_MESSAGE, payload: {message}})    
}