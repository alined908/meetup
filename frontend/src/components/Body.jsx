import React, {Component} from 'react';
import {Route, Switch} from 'react-router-dom';
import LogoutComponent from "./auth/LogoutComponent";
import ErrorComponent from "./auth/ErrorComponent";
import RegisterComponent from "./auth/RegisterComponent";
import WelcomeComponent from "./WelcomeComponent";
import ChatComponent from "./chat/ChatComponent"
import MeetupsComponent from "./meetup/MeetupsComponent";
import HomeComponent from "./HomeComponent";
import MeetupPageComponent from "./meetup/MeetupPageComponent"
import FriendsComponent from './invite/FriendsComponent';
import MeetupEventForm from "./meetup/MeetupEventForm"
import AuthenticatedRoute from "../accounts/AuthenticatedRoute";
import LoginComponent from "./auth/LoginComponent";
import Invites from "./invite/Invites"
import MeetupForm from "./meetup/MeetupForm"
import Profile from "./Profile"

class Body extends Component {

    render () {
        return (
            <div className="c">
                <Switch>
                    <Route path="/" exact component={HomeComponent}></Route>
                    <Route path="/login" component={LoginComponent}></Route>
                    <Route path="/register" component={RegisterComponent}></Route>
                    <AuthenticatedRoute path="/logout" component={LogoutComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/welcome" component={WelcomeComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/meetups/:uri/new" exact component={MeetupEventForm}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/meetups/new" exact component={MeetupForm}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/meetups/:uri" component={MeetupPageComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/meetups" exact component={MeetupsComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/chat/:uri" component={ChatComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/chat" exact component={ChatComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/friends" component={FriendsComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/invites" component={Invites}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/profile/:id" component={Profile}></AuthenticatedRoute>
                    <Route component={ErrorComponent}></Route>
                </Switch>
            </div>
        )
    }
}

export default Body