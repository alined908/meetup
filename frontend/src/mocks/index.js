import moment from 'moment'

//Category
export const category = {
    id: 1, 
    image: null,
    label: "Dessert", 
    api_label: "desserts"
}

//User
export const user = {
    id: 1, 
    email: "daniel@gmail.com", 
    first_name: "Daniel", 
    last_name: "Lee",
    avatar: null
}
export const user2 = {
    id: 2, 
    email: "example2@gmail.com", 
    first_name: "Bob", 
    last_name: "Jim", 
    avatar: null
}
export const user3 = {
    id: 3, 
    email: "example3@gmail.com", 
    first_name: "Bob", 
    last_name: "Jim", 
    avatar: null
}
export const settings = {
    radius: 25, 
    location: null, 
    latitude: null,
    longitude: null
}

export const userWithSettings = {
    ...user,
    settings: {...settings}
}
export const friend = {
    id: 1, 
    user: user2, 
    created_at: moment().format(), 
    chat_room: "something"
}
export const friends = [friend]
export const preference = {
    id: 1, 
    user, 
    category, 
    name: "Dessert", 
    ranking: 1, 
    timestamp: moment().format()
}
export const preferences = [preference]
export const friendInvite = {
    id: 1, 
    timestamp: moment().format(), 
    status: 1, 
    uri: "random", 
    sender: user, 
    receiver: user2
}

//Chat
export const message = {
    id: 1, 
    message: "hello", 
    timestamp: moment().format(), 
    room_id: 1, 
    sender: user, 
    is_notif: false
}
export const message2 = {
    id: 2, 
    message: "hello", 
    timestamp: moment().format(), 
    room_id: 1, 
    sender: user, 
    is_notif: true
}
export const messages = [message, message2]
export const members = {1: user, 2: user2}
export const room = {
    uri: {
        id: 1, 
        uri: "abc", 
        name: "Room", 
        timestamp: moment().format(), 
        members: members, 
        friendship: 1, 
        meetup:null, 
        notifs: 0
    }
}
export const rooms = {
    friend: {
        id: 1, 
        uri: "abc", 
        name: "Room", 
        timestamp: moment().format(), 
        members: members, 
        friendship: 1, 
        meetup:null, 
        notifs: 0
    },
    meetup: {
        id: 2, 
        uri: "xyz", 
        name: "Room", 
        timestamp: moment().format(), 
        members: members, 
        friendship: null, 
        meetup:1, 
        notifs: 3
    }
}

//General
export const error = {
    message: "Something went wrong."
}

//Meetup
export const meetupMember1 = {
    1: {
        id: 1,
        user,
        ban: false,
        admin: true
    }
}

export const meetupMember2 = {
    2: {
        id: 2,
        user: user2,
        ban: false,
        admin: false
    }
}

export const meetupMember3 = {
    3: {
        id: 3,
        user: user3,
        ban: false,
        admin: false
    }
}

export const meetupMembers = {...meetupMember1, ...meetupMember2}

export const meetup = {
    uri: {
        id: 1, 
        name: "Meetup", 
        uri: "uri", 
        creator: user, 
        location: "location", 
        date: moment().format("YYYY-MM-DD"),
        members: meetupMembers,
        notifs: 0,
        public: true,
        categories: [category],
        latitude: 34.228754, 
        longitude: -118.2351192,
        events: {},
        isMeetupEventsInitialized : true
    }
}

export const meetups = {...meetup}

export const event = {
    1: {
        id: 1,
        meetup: 1,
        creator: user,
        title: "Dessert",
        start: moment().format(),
        end: moment().format(),
        chosen: null,
        categories: [category],
        options: {},
        price: "1, 2",
        distance: 16000,
        entries: {},
        random: true
    }
}
export const events = {...event}

export const option = {
    1 : {
        id: 1,
        score: 0,
        option: null,
        restaurant: {},
        votes: {},
        banned: false
    }
}

export const options = {...option}

export const tokens = {
    access: "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNTg4MjI5OTEzLCJqdGkiOiIwOTg2YTg0MDA1ZGM0OGQ1ODEzYmIyNzE1OWViNGUyZCIsInVzZXJfaWQiOjEsInVzZXIiOnsiaWQiOjEsImVtYWlsIjoiZGFuaWVsQGdtYWlsLmNvbSIsImZpcnN0X25hbWUiOiJEYW5pZWwiLCJsYXN0X25hbWUiOiJMZWUiLCJhdmF0YXIiOm51bGwsInNldHRpbmdzIjp7InJhZGl1cyI6MjUsImxvY2F0aW9uIjpudWxsLCJsYXRpdHVkZSI6bnVsbCwibG9uZ2l0dWRlIjpudWxsfX19.FbffRthOKX5F0TGvqA5mZXuXUWNkUojzKexDGCoP518",
    refresh : "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoicmVmcmVzaCIsImV4cCI6MTU4ODM4Mjk0NiwianRpIjoiN2ZhNGY2MDE4OWM3NDA5ZmFjOTMwYmE4ZTRlMWFhMTMiLCJ1c2VyX2lkIjoxLCJ1c2VyIjp7ImlkIjoxLCJlbWFpbCI6ImRhbmllbEBnbWFpbC5jb20iLCJmaXJzdF9uYW1lIjoiRGFuaWVsIiwibGFzdF9uYW1lIjoiTGVlIiwiYXZhdGFyIjpudWxsLCJzZXR0aW5ncyI6eyJyYWRpdXMiOjI1LCJsb2NhdGlvbiI6bnVsbCwibGF0aXR1ZGUiOm51bGwsImxvbmdpdHVkZSI6bnVsbH19fQ.qqaY4Ye7rmyOKLbDS2EKz4SGWH0fwZ0MvrGP6NJ3IRE"
}