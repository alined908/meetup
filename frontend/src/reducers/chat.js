import {GET_ROOMS_REQUEST, GET_ROOMS_SUCCESS, GET_ROOMS_ERROR, 
    GET_MESSAGES_REQUEST, GET_MESSAGES_SUCCESS, GET_MESSAGES_ERROR, 
    GET_MORE_MESSAGES_REQUEST, GET_MORE_MESSAGES_SUCCESS, GET_MORE_MESSAGES_ERROR,
    CLEAR_STORE, REMOVE_ACTIVE_ROOM, ADD_ROOM, SET_ACTIVE_ROOM, SET_TYPING_VALUE, ADD_MESSAGE} from "../constants/action-types"

const defaultState = {
    rooms: {},
    messages: [],
    activeRoom: null,
    setTypingValue: "",
    isActiveRoomSet: false,
    isRoomsInitialized: false,
    isRoomsFetching: false,
    isMessagesInitialized: false,
    isMessagesFetching: false,
    isMoreMessagesFetching: false
}

export default function chatReducer(state = defaultState, action){
    switch(action.type){
        case GET_ROOMS_REQUEST:
            return {...state, isRoomsFetching: true}
        case GET_ROOMS_SUCCESS:
            return {...state, rooms: action.payload, isRoomsInitialized: true, isRoomsFetching: false}
        case GET_ROOMS_ERROR:
            return {...state, isRoomsFetching: false, errorMessage: action.payload.message}
        case GET_MESSAGES_REQUEST:
            return {...state, isMessagesFetching: true}
        case GET_MESSAGES_SUCCESS:
            return {...state, messages: action.payload, isMessagesInitialized: true, isMessagesFetching: false}
        case GET_MESSAGES_ERROR:
            return {...state, isMessagesFetching: false, errorMessage: action.payload.message}
        case GET_MORE_MESSAGES_REQUEST:
            return {...state, isMoreMessagesFetching: true}
        case GET_MORE_MESSAGES_SUCCESS:
            return {...state, messages: [...action.payload, ...state.messages], isMoreMessagesFetching: false}
        case GET_MORE_MESSAGES_ERROR:
            return {...state, errorMessage: action.payload.message, isMoreMessagesFetching: false}
        case ADD_ROOM:
            return {...state, rooms: state.rooms.concat([action.payload])}
        case SET_ACTIVE_ROOM:
            return {...state, activeRoom: action.payload.uri, isActiveRoomSet: true, isMessagesInitialized: false}
        case REMOVE_ACTIVE_ROOM:
            return {...state, activeRoom: null, isActiveRoomSet:false, isMessagesInitialized:false}
        case SET_TYPING_VALUE:
            return {...state, setTypingValue: action.payload}
        case ADD_MESSAGE:
            return {...state, setTypingValue: '', messages: [...state.messages, action.payload.message]}
        case CLEAR_STORE:
            return defaultState
        default:
            return state
    }
}