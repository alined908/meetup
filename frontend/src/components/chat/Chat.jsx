import React, { Component } from 'react'
import { ChatBar, ChatWindow } from "../components"
import { connect } from "react-redux";
import {getMessages, setActiveRoom, addMessage, removeActiveRoom, getRooms} from "../../actions"
import WebSocketService from "../../accounts/WebSocket"
import AuthenticationService from '../../accounts/AuthenticationService';
import { Helmet } from "react-helmet"
import styles from '../../styles/chat.module.css'

class Chat extends Component {
    constructor(props){
        super(props)
        this.state = {
            socket: new WebSocketService(),
            nonMobile: window.matchMedia("(min-width: 768px)").matches
        }
        this.state.socket.addChatCallbacks(this.props.getMessages, this.props.addMessage)
    }

    componentDidMount(){
        const handler = e => this.setState({nonMobile: e.matches});
        window.matchMedia("(min-width: 768px)").addListener(handler);
        this.props.getRooms()
        if("uri" in this.props.match.params){
            this.getRelevantInfo(this.props.match.params.uri)
        }
    }

    componentDidUpdate(prevProps){
        if (this.props.match.params.uri !== prevProps.match.params.uri){
            if (this.state.socket.exists()){
                this.state.socket.disconnect()
            }
            this.getRelevantInfo(this.props.match.params.uri)
        }
    }

    componentWillUnmount(){
        this.props.removeActiveRoom()
        this.state.socket.disconnect()
    }

    getRelevantInfo(uri) {
        const socket = this.state.socket
        const token = AuthenticationService.retrieveToken()
        const path = `/ws/chat/${this.props.match.params.uri}/`;
        socket.connect(path, token);
        this.setRoomInfo(uri)
    }

    setRoomInfo(uri){
        this.props.setActiveRoom(uri);
        this.props.getMessages(uri);
    }

    render(){
        const renderChatWindow = () => {
            if (this.props.isMessagesInitialized || this.props.isMessagesFetching) {
                return <ChatWindow
                            socket={this.state.socket} isMessagesInitialized={this.props.isMessagesInitialized} 
                            activeRoom={this.props.activeRoom} messages={this.props.messages}
                        />
            } else {
                return <ChatWindow socket={this.state.socket} activeRoom={null} messages={[]}/>
            }
        }

        return (
            <div className={`${styles.chat}` + (this.state.nonMobile ? "" : "chat-mobile")}>
                <Helmet>
                    <meta charSet="utf-8" />
                    <title>Chat</title>
                    <meta name="description" content="Phoodie Chat" />
                </Helmet>
                <ChatBar rooms={this.props.rooms}/>
                {renderChatWindow()}
            </div>
        )
    }
}

const mapDispatchToProps = {
    setActiveRoom,
    getMessages,
    getRooms,
    addMessage,
    removeActiveRoom
}

function mapStateToProps(state){
    return {
        isMessagesInitialized: state.chat.isMessagesInitialized,
        activeRoom: state.chat.activeRoom,
        messages: state.chat.messages,
        isActiveRoomSet: state.chat.isActiveRoomSet,
        rooms: Object.values(state.chat.rooms),
        isMessagesFetching: state.chat.isMessagesFetching
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Chat)
export {Chat as UnderlyingChat}