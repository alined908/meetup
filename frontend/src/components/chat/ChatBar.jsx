import React, {Component} from 'react';
import {Contact} from "../components";
import {connect} from 'react-redux';
import {IconButton, Tooltip, CircularProgress} from '@material-ui/core'
import {Event as EventIcon, Person as PersonIcon, Search as SearchIcon, Error as ErrorIcon} from '@material-ui/icons';
import PropTypes from 'prop-types';
import {chatRoomPropType, userPropType,} from "../../constants/prop-types"
import WebSocketService from "../../accounts/WebSocket"
import AuthenticationService from '../../accounts/AuthenticationService';
import {updateRoom} from '../../actions'
import styles from '../../styles/chat.module.css'

class ChatBar extends Component {
    constructor(props){
        super(props)
        this.state = {
            rooms: [],
            filteredRooms: [],
            searchedRooms: [],
            liveRooms: [],
            filters: {
                "friend": false, 
                "meetup": false
            },
            isFiltered: false,
            isSearched: false,
            searchInput: '',
            socket: new WebSocketService()
        }
        this.state.socket.addContactsCallbacks(this.props.updateRoom)
    }

    // Initialize socket for chat rooms.
    componentDidMount(){
        const socket = this.state.socket
        const token = AuthenticationService.retrieveToken()
        const path = `/ws/chat/rooms/`;
        socket.connect(path, token);
    }

    componentDidUpdate(prevProps) {
        if (this.props.rooms !== prevProps.rooms) {
            this.setState({rooms: this.props.rooms, liveRooms: this.props.rooms})
        }
    }

    filterAndSearch = () => {
        const rooms = this.state.rooms
        const filter = this.state.searchInput
        const filterMeetup = this.state.filters["meetup"]
        const filterFriend = this.state.filters["friend"]
        let newRooms;

        newRooms = rooms.filter((room) => {
        
            let meetupCriteria = false;
            if (room.meetup !== null) {
                meetupCriteria = room.name.toLowerCase().includes(filter.toLowerCase()) && !filterMeetup
            }

            let friendCriteria = false;
            if (room.friendship !== null){
                const user = this.props.user
                let friend;
                for (let key in room.members){
                    if (key !== user.id.toString()){
                        friend = room.members[key]
                    }
                }
                const friendName = `${friend.first_name} ${friend.last_name}`

                friendCriteria = (
                    (friendName.toLowerCase().includes(filter.toLowerCase())) || 
                    (friend.email.toLowerCase().includes(filter.toLowerCase()))
                ) && !filterFriend
            }

            return meetupCriteria || friendCriteria
        })

        this.setState({liveRooms: newRooms})
    }

    filterRooms = (type) => {
        let filters = {...this.state.filters}
        filters[type] = !filters[type]
        this.setState({filters}, () => this.filterAndSearch())
    }

    handleSearchInputChange = (e) => {
        var filter = e.target.value;
        this.setState({searchInput: filter}, () => this.filterAndSearch())
    }

    render (){
        const rooms = this.state.liveRooms

        return (
            <div className={styles.bar}>
                <div className={styles.barTop}>
                    <div>Contacts</div>
                    <div>
                        <Tooltip title="Friends">
                            <IconButton 
                                color={this.state.filters["friend"] ? "default" : "primary"} 
                                onClick={() => this.filterRooms("friend")} edge="end"
                            >
                                <PersonIcon/>
                            </IconButton>
                        </Tooltip>
                        <Tooltip title="Meetups">
                            <IconButton 
                                color={this.state.filters["meetup"] ? "default" : "primary"} 
                                onClick={() => this.filterRooms("meetup")}
                            >
                                <EventIcon/>
                            </IconButton>
                        </Tooltip>
                    </div>
                </div>
                <div className={styles.contacts}>
                    {this.props.isRoomsFetching && 
                        <div className="loading">
                            <CircularProgress/>
                        </div>
                    }
                    {(this.props.isRoomsInitialized && rooms.length === 0) && 
                        <div className="no-entity">
                            <ErrorIcon style={{color: "rgb(255, 212, 96)"}}/>
                            <span className="no-entity-text">
                                No chat rooms match criteria!
                            </span>
                        </div>
                    }
                    {this.props.isRoomsInitialized && rooms.map((room) => 
                        <Contact 
                            key={room.id}
                            room={room}
                            user={this.props.user} 
                            currentRoom={this.props.currentRoom}
                        />
                    )}
                </div>
                <div className={styles.search}>
                    <SearchIcon/>
                    <input 
                        type="text"
                        className={styles.input} 
                        placeholder="Search Contacts..." 
                        value={this.state.searchInput} 
                        onChange={this.handleSearchInputChange}
                    />
                </div>
            </div>
        )
    }
}

ChatBar.propTypes = {
    rooms: PropTypes.arrayOf(
        chatRoomPropType
    ),
    user: userPropType,
    isRoomsInitialized: PropTypes.bool
}

const mapStateToProps = (state) => {
    return {
        user: state.user.user,
        isRoomsInitialized: state.chat.isRoomsInitialized,
        isRoomsFetching: state.chat.isRoomsFetching,
        currentRoom: state.chat.activeRoom
    }
}

const mapDispatchToProps = {
    updateRoom
}

export default connect(mapStateToProps, mapDispatchToProps)(ChatBar)
export {ChatBar as UnderlyingChatBar}