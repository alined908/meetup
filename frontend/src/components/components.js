//Auth
import ErrorComponent from "./auth/ErrorComponent";
import LoginComponent from "./auth/LoginComponent";
import LogoutComponent from "./auth/LogoutComponent";
import RegisterComponent from "./auth/RegisterComponent";

//Chat
import ChatBarComponent from "./chat/ChatBarComponent"
import ChatComponent from "./chat/ChatComponent"
import ChatMessageComponent from "./chat/ChatMessageComponent"
import ChatWindowComponent from "./chat/ChatWindowComponent"
import ContactComponent from "./chat/ContactComponent"

//invite
import Friend from './invite/Friend';
import FriendsComponent from './invite/FriendsComponent';
import Invite from "./invite/Invite"
import Invites from "./invite/Invites"

//meetup
import Map from "./meetup/Map"
import Meetup from "./meetup/Meetup";
import MeetupCard from "./meetup/MeetupCard";
import MeetupEvent from "./meetup/MeetupEvent";
import MeetupEventForm from "./meetup/MeetupEventForm"
import MeetupForm from "./meetup/MeetupForm";
import MeetupFriend from "./meetup/MeetupFriend";
import MeetupPageComponent from "./meetup/MeetupPageComponent"
import MeetupsComponent from "./meetup/MeetupsComponent";
import Restauraunt from "./meetup/Restauraunt";

//fields
import CategoryAutocomplete from "./fields/CategoryAutocomplete"
import Location from "./fields/Location"
import renderTextField from "./fields/renderTextField"
import renderDatePicker from "./fields/renderDatePicker"
import renderDateSimplePicker from "./fields/renderDateSimplePicker"
import UserAutocomplete from "./fields/UserAutocomplete"

//General
import Body from "./general/Body"
import GlobalMessage from "./general/GlobalMessage"
import HomeComponent from "./general/HomeComponent";
import LiveUpdatingBadge from "./general/LiveUpdatingBadge"
import Navigation from "./general/Navigation"
import Profile from "./general/Profile"
import GroupAvatars from './general/GroupAvatars'
import NotificationWrapper from "./general/NotificationWrapper"
import CalendarComponent from "./general/CalendarComponent"
import CategoriesComponent from "./general/CategoriesComponent"
import CategoryComponent from "./general/CategoryComponent"
import Preferences from "./general/Preferences"
import Preference from "./general/Preference"
import SettingsComponent from "./general/SettingsComponent"

export {
    ErrorComponent, LoginComponent, LogoutComponent, RegisterComponent, 
    ChatBarComponent, ChatComponent, ChatMessageComponent, ChatWindowComponent, ContactComponent,
    Friend, FriendsComponent, Invite, Invites,
    Map, Meetup, MeetupCard, MeetupEvent, MeetupEventForm, MeetupFriend, MeetupForm, MeetupPageComponent, MeetupsComponent, Restauraunt,
    CategoryAutocomplete, UserAutocomplete, Location, renderTextField, renderDatePicker, renderDateSimplePicker,
    Body, GlobalMessage, GroupAvatars, HomeComponent, LiveUpdatingBadge, Navigation, Profile, NotificationWrapper, 
    Preferences, Preference, CalendarComponent, CategoriesComponent, CategoryComponent, SettingsComponent}