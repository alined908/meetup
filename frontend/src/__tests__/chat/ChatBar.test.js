import React from 'react'
import {shallow} from 'enzyme';
import {ChatBar} from "../../components/components" 
import {UnderlyingChatBar} from "../../components/chat/ChatBar"
import {user, user2, rooms} from "../../mocks/index"

describe("ChatBar unit", () => {

    it ('renders without crashing given props', () => {
        const props = {
            rooms: Object.values(rooms)
        }
        const wrapper = shallow(<UnderlyingChatBar {...props}/>)
        expect(wrapper).toMatchSnapshot()
    })

    it ('correctly displays message when there are no chat rooms', () => {
        const props = {
            rooms: [],
            user: user,
            isRoomsInitialized: true
        }
        const wrapper = shallow(<UnderlyingChatBar {...props}/>)
        expect(wrapper.find(".chat-bar-none")).toHaveLength(1)
    })

    it ('correctly filters out rooms onclick', () => {
        const props = {
            rooms: [],
            user: user,
            isRoomsInitialized: true
        }
        const wrapper = shallow(<UnderlyingChatBar {...props}/>)
        const friendButton = wrapper.find("WithStyles(ForwardRef(IconButton))").first()
        const meetupButton = wrapper.find("WithStyles(ForwardRef(IconButton))").at(1)
        wrapper.setProps({rooms: Object.values(rooms)})
        expect(wrapper.find("Connect(Contact)")).toHaveLength(2)
        friendButton.simulate('click')
        expect(wrapper.find("Connect(Contact)")).toHaveLength(1)
        expect(wrapper.find("Connect(Contact)").first().props().room.uri).toEqual(rooms.meetup.uri)
        friendButton.simulate('click')
        expect(wrapper.find("Connect(Contact)")).toHaveLength(2)
        meetupButton.simulate('click')
        expect(wrapper.find("Connect(Contact)")).toHaveLength(1)
        expect(wrapper.find("Connect(Contact)").first().props().room.uri).toEqual(rooms.friend.uri)
        meetupButton.simulate('click')
        expect(wrapper.find("Connect(Contact)")).toHaveLength(2)
        meetupButton.simulate('click')
        friendButton.simulate('click')
        expect(wrapper.find("Connect(Contact)")).toHaveLength(0)
    })

    it ('correctly filters out rooms onsearch', () => {
        const props = {
            rooms: [],
            user: user,
            isRoomsInitialized: true
        }
        const wrapper = shallow(<UnderlyingChatBar {...props}/>)
        const chatInput = wrapper.find(".chat-input")
        wrapper.setProps({rooms: Object.values(rooms)})
        chatInput.simulate("change", {target: {value: "me"}})
        expect(wrapper.state('searchInput')).toEqual("me")
        expect(wrapper.find("Connect(Contact)")).toHaveLength(1)
        expect(wrapper.find("Connect(Contact)").first().props().room.uri).toEqual(rooms.meetup.uri)
        chatInput.simulate("change", {target: {value: user2.first_name}})
        expect(wrapper.state('searchInput')).toEqual(user2.first_name)
        expect(wrapper.find("Connect(Contact)")).toHaveLength(1)
        expect(wrapper.find("Connect(Contact)").first().props().room.uri).toEqual(rooms.friend.uri)
    })

    it ('correctly filters out rooms onsearch and onclick', () => {
        const props = {
            rooms: [],
            user: user,
            isRoomsInitialized: true
        }
        const wrapper = shallow(<UnderlyingChatBar {...props}/>)
        const friendButton = wrapper.find("WithStyles(ForwardRef(IconButton))").first()
        const meetupButton = wrapper.find("WithStyles(ForwardRef(IconButton))").at(1)
        const chatInput = wrapper.find(".chat-input")
        wrapper.setProps({rooms: Object.values(rooms)})
        chatInput.simulate("change", {target: {value: "me"}})
        expect(wrapper.find("Connect(Contact)")).toHaveLength(1)
        meetupButton.simulate('click')
        expect(wrapper.find("Connect(Contact)")).toHaveLength(0)
        chatInput.simulate("change", {target: {value: user2.first_name}})
        expect(wrapper.state('searchInput')).toEqual(user2.first_name)
        expect(wrapper.state('filters')).toEqual({"friend": false, "meetup": true})
        expect(wrapper.find("Connect(Contact)")).toHaveLength(1)
        friendButton.simulate('click')
        expect(wrapper.find("Connect(Contact)")).toHaveLength(0)
        chatInput.simulate("change", {target: {value: ""}})
        expect(wrapper.find("Connect(Contact)")).toHaveLength(0)
    })
})