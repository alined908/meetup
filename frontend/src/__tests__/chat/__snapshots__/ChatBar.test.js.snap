// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatBarComponent unit renders without crashing given props 1`] = `
<div
  className="chat-bar elevate"
>
  <div
    className="chat-bar-top"
  >
    <div>
      Contacts
    </div>
    <div>
      <WithStyles(ForwardRef(Tooltip))
        title="Friends"
      >
        <WithStyles(ForwardRef(IconButton))
          color="primary"
          edge="end"
          onClick={[Function]}
        >
          <PersonIcon />
        </WithStyles(ForwardRef(IconButton))>
      </WithStyles(ForwardRef(Tooltip))>
      <WithStyles(ForwardRef(Tooltip))
        title="Meetups"
      >
        <WithStyles(ForwardRef(IconButton))
          color="primary"
          onClick={[Function]}
        >
          <EventIcon />
        </WithStyles(ForwardRef(IconButton))>
      </WithStyles(ForwardRef(Tooltip))>
    </div>
  </div>
  <div
    className="chat-bar-contacts"
  >
    <div>
      Initializing Contacts...
    </div>
  </div>
  <div
    className="chat-bar-search"
  >
    <SearchIcon />
    <input
      className="chat-input"
      onChange={[Function]}
      placeholder="Search Contacts..."
      type="text"
      value=""
    />
  </div>
</div>
`;
